tests/DCPS/EntityLifecycleStress/publisher.cpp

```diff
@@ -0,0 +1,137 @@
+  try
+    {
+      DDS::DomainParticipantFactory_var dpf =
+        TheParticipantFactoryWithArgs(argc, argv);
+
+      DDS::DomainParticipant_var participant =
+        dpf->create_participant(31,
+                                PARTICIPANT_QOS_DEFAULT,
+                                DDS::DomainParticipantListener::_nil(),
+                                ::OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+      if (CORBA::is_nil(participant.in())) {
+        cerr << "create_participant failed." << endl;
+        return 1;
+      }
+
+      MessageTypeSupportImpl::_var_type servant = new MessageTypeSupportImpl();
+
+      if (DDS::RETCODE_OK != servant->register_type(participant.in(), "")) {
+        cerr << "register_type failed." << endl;
+        exit(1);
+      }
+
+      CORBA::String_var type_name = servant->get_type_name();
+
+      DDS::TopicQos topic_qos;
+      participant->get_default_topic_qos(topic_qos);
+      DDS::Topic_var topic =
+        participant->create_topic("Movie Discussion List",
+                                  type_name.in(),
+                                  topic_qos,
+                                  DDS::TopicListener::_nil(),
+                                  ::OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+
+      if (CORBA::is_nil(topic.in())) {
+        cerr << "create_topic failed." << endl;
+        exit(1);
+      }
+
+      DDS::PublisherQos pub_qos;
+      participant->get_default_publisher_qos(pub_qos);
+
+      DDS::Publisher_var pub =
+        participant->create_publisher(pub_qos, DDS::PublisherListener::_nil(),
+                                      ::OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+      if (CORBA::is_nil(pub.in())) {
+        cerr << "create_publisher failed." << endl;
+        exit(1);
+      }
+
+      DDS::DataWriter_var dw =
+        pub->create_datawriter(topic.in(),
+                               DATAWRITER_QOS_DEFAULT,
+                               DDS::DataWriterListener::_nil(),
+                               ::OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+
+      Messenger::MessageDataWriter_var message_dw =
+        Messenger::MessageDataWriter::_narrow(dw.in());
+
+      Messenger::Message message;
+      message.subject_id = 16;
+
+      ::DDS::InstanceHandle_t handle = message_dw->register_instance(message);
+
+      {
+        stringstream ss;
+        ss << ACE_OS::getpid() << flush;
+
+        message.from       = ss.str().c_str();
+        message.subject    = CORBA::string_dup("Review");
+        message.text       = CORBA::string_dup("Worst. Movie. Ever.");
+        message.count      = 0;
+      }
+
+      const ACE_Time_Value delay(0, 10000);
+
+      const size_t count = 250u;
+      for (size_t i = 0; i < count; ++i) {
+        ACE_OS::sleep(delay);
+        ++message.count;
+        message_dw->write(message, handle);
+      }
+
+      {
+        stringstream ss;
+        ss << "Publisher " << ACE_OS::getpid() << " is done. Exiting." << endl;
+
+        cout << ss.str() << flush;
```

Comment:
Why does this need a separate `stringstream`?

Reply:
I can't remember if it's only a problem on some platforms and not others, but multi-process IO to the console can potentially switch processes between << operators unless it's done as a single output stream. So for the output of these tests I tried to protect against it with stringstream.